agent.go

```diff
@@ -127,12 +127,15 @@ type agent interface {
 	// start will start the agent on the host.
 	start() error
 
+	// isStarted will check the agent is started.
+	isStarted() bool
```

Comment:
I'd prefer that one to be more generic: `verifyState() agentStateString`. And we should update and potentially store the new agent state based as part of that function.